src/webots/engine/WbSimulationCluster.cpp

```diff
@@ -455,6 +456,11 @@ static bool needCollisionDetection(WbSolid *solid, bool isOtherRayGeom) {
   return false;
 }
 
+void WbSimulationCluster::appendCollisionedRobot(WbKinematicDifferentialWheels *robot) {
+  QMutexLocker lock(&mCollisionedRobotsMutex);
+  mCollisionedRobots.append(robot);
```

Comment:
Did you evaluate the performance impact of this? Is it worth to add some quick test like:
```suggestion
  if (WbOdeContext::instance()->numberOfThreads() > 1) {
    QMutexLocker lock(&mCollisionedRobotsMutex);
    mCollisionedRobots.append(robot);
  } else
    mCollisionedRobots.append(robot);
```

Reply:
I didn't see a noticeable performance degradation, but my testing was not exhaustive and your suggestion adds minimal complexity.